homeassistant/components/onkyo/media_player.py

```diff
@@ -106,462 +203,405 @@
-        # the display shows 'Not Available'.
-        # We avoid this by checking if HDMI out is supported
-        if self._hdmi_out_supported:
-            hdmi_out_raw = self.command("hdmi-output-selector query")
-        else:
-            hdmi_out_raw = []
-        preset_raw = self.command("preset query")
-        if self._audio_info_supported:
-            audio_information_raw = self.command("audio-information query")
-            self._parse_audio_information(audio_information_raw)
-        if self._video_info_supported:
-            video_information_raw = self.command("video-information query")
-            self._parse_video_information(video_information_raw)
-        if not (volume_raw and mute_raw and current_source_raw):
-            return
+    async def async_added_to_hass(self) -> None:
+        """Entity has been added to hass."""
+        self.backfill_state()
 
-        sources = _parse_onkyo_payload(current_source_raw)
+    async def async_will_remove_from_hass(self) -> None:
+        """Cancel the query timer when the entity is removed."""
+        if self._query_timer:
+            self._query_timer()
+            self._query_timer = None
 
-        for source in sources:
-            if source in self._source_mapping:
-                self._attr_source = self._source_mapping[source]
-                break
-            self._attr_source = "_".join(sources)
+    @property
+    def supported_features(self) -> MediaPlayerEntityFeature:
+        """Return media player features that are supported."""
+        if self._supports_sound_mode:
+            return SUPPORT_ONKYO
+        if self._supports_volume:
+            return SUPPORT_ONKYO_WO_SOUND_MODE
+        return SUPPORT_ONKYO_WO_VOLUME
 
-        if preset_raw and self.source and self.source.lower() == "radio":
-            self._attr_extra_state_attributes[ATTR_PRESET] = preset_raw[1]
-        elif ATTR_PRESET in self._attr_extra_state_attributes:
-            del self._attr_extra_state_attributes[ATTR_PRESET]
+    @callback
+    def _update_receiver(self, propname, value):
+        """Update a property in the receiver."""
+        self._receiver.update_property(self._zone, propname, value)
 
-        self._attr_is_volume_muted = bool(mute_raw[1] == "on")
-        # AMP_VOL / (MAX_RECEIVER_VOL * (MAX_VOL / 100))
-        self._attr_volume_level = volume_raw[1] / (
-            self._receiver_max_volume * self._max_volume / 100
-        )
+    @callback
+    def _query_receiver(self, propname):
+        """Cause the receiver to send an update about a property."""
+        self._receiver.query_property(self._zone, propname)
 
-        if not hdmi_out_raw:
-            return
-        self._attr_extra_state_attributes[ATTR_VIDEO_OUT] = ",".join(hdmi_out_raw[1])
-        if hdmi_out_raw[1] == "N/A":
-            self._hdmi_out_supported = False
+    async def async_turn_on(self) -> None:
+        """Turn the media player on."""
+        self._update_receiver("power", "on")
```

Comment:
can we use consts for these strings?

Reply:
That is the way it was before as well, maybe let's address it in the follow up PR.